File: 58-length-of-last-word/length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        # return len(s.strip().split(" ")[-1])
        # -------------------- OR ----------------------

        res = 0
        for i in range(len(s)-1, -1, -1):
            if s[i] != " ":
                res += 1
            elif res:
                return res
        return res
        

File: 58-length-of-last-word/length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int res = 0;
        char[] arr = s.toCharArray();
        for (int i = arr.length - 1; i >= 0; i--) {
            if (arr[i] != ' ') {
                res++;
            } else if (res > 0) {
                return res;
            }
        }
        return res;
    }
}
